pandas/core/arrays/array_.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+
+from pandas._libs import lib, tslibs
+
+from pandas.core.dtypes.common import is_extension_array_dtype
+from pandas.core.dtypes.dtypes import registry
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+
+from pandas import compat
+
+
+def array(data, dtype=None, copy=False):
+    """
+    Create an array.
+
+    .. versionadded:: 0.24.0
+
+    Parameters
+    ----------
+    data : Sequence[object]
+        A sequence of objects. The scalars inside `data` should
+        be instances of the scalar type for `dtype`.
+
+        When `data` is an Index or Series, the underlying array
+        will be extracted from `data`.
+
+    dtype : Union[str, np.dtype, ExtensionDtype], optional
+        The dtype to use for the array. This may be a NumPy
+        dtype or an extension type registered with pandas using
+        :meth:`pandas.api.extensions.register_extension_dtype`.
+
+        If not specified, there are two possibilities:
+
+        1. When `data` is a :class:`Series`, :class:`Index`, or
+           :class:`ExtensionArray`, the `dtype` will be taken
+           from the data.
```

Comment:
also when `data` is a numpy array I assume?

Reply:
Ah, I see the note a couple of lines below. However, I don't think that is completely true:

```
In [1]: pd.array(np.array([1, 2], dtype='int32'))                                                                                                                                                                   
Out[1]: array([1, 2], dtype=int32)

In [2]: pd.array([1, 2])                                                                                                                                                                                            
Out[2]: array([1, 2])
```

Maybe a logical example, but so strictly spoken, `data.dtype` is not ignored here in that first case.

Reply:
I've clarified that `data.dtype` is ignored only when it comes to inference.